src/server/scripts/Spells/spell_monk.cpp

```diff
@@ -186,9 +190,172 @@ class spell_monk_provoke : public SpellScriptLoader
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_MONK_RENEWING_MIST_PERIODIC_HEAL))
+                    return false;
+                if (!sSpellMgr->GetSpellInfo(SPELL_MONK_RENEWING_MIST_TARGET_SELECT))
+                    return false;
+                if (!sSpellMgr->GetSpellInfo(SPELL_MONK_RENEWING_MIST_DUMMY_VISUAL))
+                    return false;
+                return true;
+            }
+
+            void HandlePeriodic(AuraEffect const* aurEff)
+            {
+                if (aurEff->GetBase()->GetCharges() > 1)
+                    if (Unit* originCaster = GetCaster())
+                        originCaster->CastSpell(GetTarget(), SPELL_MONK_RENEWING_MIST_TARGET_SELECT, true);
+            }
+
+            void HandleDummy(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
+            {
+                aurEff->GetBase()->SetCharges(aurEff->GetAmount());
+            }
+
+            void Register() override
+            {
+                OnEffectPeriodic += AuraEffectPeriodicFn(spell_monk_renewing_mist_periodic_heal_AuraScript::HandlePeriodic, EFFECT_0, SPELL_AURA_PERIODIC_HEAL);
+                OnEffectApply += AuraEffectApplyFn(spell_monk_renewing_mist_periodic_heal_AuraScript::HandleDummy, EFFECT_1, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+            }
+        };
+
+        AuraScript* GetAuraScript() const override
+        {
+            return new spell_monk_renewing_mist_periodic_heal_AuraScript();
+        }
+};
+
+// 119607 - Renewing Mist Target Selection
+class spell_monk_renewing_mist_target_selector : public SpellScriptLoader
+{
+    public:
+        spell_monk_renewing_mist_target_selector() : SpellScriptLoader("spell_monk_renewing_mist_target_selector") { }
+
+        class spell_monk_renewing_mist_target_selector_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_monk_renewing_mist_target_selector_SpellScript);
+
+        public:
+            spell_monk_renewing_mist_target_selector_SpellScript()
+            {
+                emptyTargets = false;
+            }
+
+        private:
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_MONK_RENEWING_MIST_PERIODIC_HEAL))
+                    return false;
+                if (!sSpellMgr->GetSpellInfo(SPELL_MONK_RENEWING_MIST_TARGET_SELECT))
+                    return false;
+                if (!sSpellMgr->GetSpellInfo(SPELL_MONK_RENEWING_MIST_DUMMY_VISUAL))
+                    return false;
+                return true;
+            }
+
+            void FilterTargets(std::list<WorldObject*>& targets)
+            {
+                if (SpellEffectInfo const* eff = GetSpellInfo()->GetEffect(EFFECT_1))
+                {
+                    targets.remove(GetExplTargetWorldObject());
+                    if (targets.empty())
+                    {
+                        emptyTargets = true;
```

Comment:
Is that bool really needed? OnEffectHitTarget is meant to be executed after hitting someone from targets, so I assume it won't be null in HandleDummy. Tell me if you tested without this check and what happened just to know for the next time.